fem/fe/fe_rbf.hpp

```diff
@@ -0,0 +1,551 @@
+// Copyright (c) 2010-2023, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_FE_RBF
+#define MFEM_FE_RBF
+
+#include "fe_base.hpp"
+
+namespace mfem
+{
+
+// Pure virtual class for dimensionless radial basis functions
```

Comment:
Do you assume that the support is always spherical? Can we have a box support and shape functions obtained as Cartesian product of 1D shape functions?

Reply:
For now, the functions assume a grid of points in the appropriate dimension. There is a choice of distance metric here: https://github.com/mfem/mfem/blob/meshless-kernels/fem/fe/fe_rbf.hpp#L277-L347. With the L1 distance, the functions are box-shaped. Unfortunately, the evaluation would not use a product of 1D shape functions. It would be good to explore that in the future, as such functions would be better suited for partial assembly.